mantle/platform/api/openstack/api.go

```diff
@@ -81,6 +81,18 @@ type API struct {
 }
 
 func New(opts *Options) (*API, error) {
+	// The clientconfig library tries to find a clouds.yaml in:
+	//     1. OS_CLIENT_CONFIG_FILE
+	//     2. Current directory.
+	//     3. unix-specific user_config_dir (~/.config/openstack/clouds.yaml)
+	//     4. unix-specific site_config_dir (/etc/openstack/clouds.yaml)
+	// See https://github.com/gophercloud/utils/blob/8677e053dcf1f05d0fa0a616094aace04690eb94/openstack/clientconfig/utils.go#L93-L112
+	//
+	// If the user provided a path to a config file set the
+	// $OS_CLIENT_CONFIG_FILE env var to it.
+	if opts.ConfigPath != "" {
+		os.Setenv("OS_CLIENT_CONFIG_FILE", opts.ConfigPath)
```

Comment:
See https://github.com/coreos/coreos-assembler/pull/1752#discussion_r505846925 for discussions about `Setenv`. In this case, since this is inside the API already, it seems even more valid (i.e. the caller might've spawned a bunch of threads already).

Hmm, it seems odd to have to communicate with the API via env vars like this. Is there really no way to tell it what config path to use?

Reply:
I agree. I looked but didn't see a clear way. Let me know if you see something different in https://github.com/gophercloud/utils/blob/8677e053dcf1f05d0fa0a616094aace04690eb94/openstack/clientconfig/

Reply:
Hmm OK, if I'm reading https://pkg.go.dev/github.com/gophercloud/utils@v0.0.0-20201101202656-8677e053dcf1/openstack/clientconfig#ClientOpts right, I think one can set `YAMLOpts` in the `ClientOpts{}` to a type which implements `LoadCloudsYAML()`. Kind of heavyweight though.

Anyway, if you've tested this and it works and you don't see a simpler way to not use `Setenv`, SGTM. Though let's revisit this soon-ish?

Reply:
> Hmm OK, if I'm reading https://pkg.go.dev/github.com/gophercloud/utils@v0.0.0-20201101202656-8677e053dcf1/openstack/clientconfig#ClientOpts right, I think one can set `YAMLOpts` in the `ClientOpts{}` to a type which implements `LoadCloudsYAML()`. Kind of heavyweight though.

Yeah. Heavyweight. 

> 
> Anyway, if you've tested this and it works and you don't see a simpler way to not use `Setenv`, SGTM. Though let's revisit this soon-ish?

Yep. Tested this. As far as I'm concerned if it's not easier via their API then using `Setenv` should be fine. Is there a good reason to revisit? I guess one option would be for us to request upstream for them to make it easier.

Reply:
We could probably just implement that interface on the `Options` type we have here, which already has the `ConfigPath` in it. Then you'd do something basically like this: https://github.com/gophercloud/utils/blob/8677e053dcf1f05d0fa0a616094aace04690eb94/openstack/clientconfig/requests.go#L92-L108. Except that for `LoadCloudsYAML()`, you'd first check if your `ConfigPath` is set before falling back to the default implementation.

Reply:
> Is there a good reason to revisit?

Yes, undefined behaviour if other threads are also doing `getenv()` at the same time.

Reply:
Heh OK this was easier than I expected: https://github.com/coreos/coreos-assembler/pull/1845. Would you be able to test that one with your OpenStack `clouds.yaml`?